src/methods.jl

```diff
@@ -59,15 +62,20 @@ In-place version of [`subtract_overscan`](@ref)
 # See Also
 [`subtract_overscan`](@ref)
 """
-function subtract_overscan!(frame::AbstractArray, idxs; dims = axes_min_length(idxs))
+function subtract_overscan!(frame::AbstractArray{T}, idxs; dims = axes_min_length(idxs)) where T
     overscan_region = @view frame[idxs...]
-    overscan_value = median(overscan_region, dims = dims)
+    overscan_value = convert_value.(median(overscan_region, dims = dims), T)
     frame .-= overscan_value
     return frame
 end
 
 subtract_overscan!(frame::AbstractArray, idxs::String; kwargs...) = subtract_overscan!(frame, fits_indices(idxs); kwargs...)
 
+# support for frame as strings(Location of file) and ImageHDU
+subtract_overscan!(frame::ImageHDU, key::Symbol; kwargs...) = subtract_overscan!(read(frame), read_header(frame)[string(key)]; kwargs...)
+subtract_overscan!(frame::ImageHDU, idxs; kwargs...) = subtract_overscan!(read(frame), idxs; kwargs...)
+subtract_overscan!(filename::String, idxs; hdu = 1, kwargs...) = subtract_overscan!(FITS(filename)[hdu], idxs; kwargs...)
```

Comment:
Missing something here? :wink:

Reply:
Uh, looking better, probably not!

Reply:
> Missing something here?

Are you referring to dispatch for `frame` as `FITS`?

Reply:
No, the semicolon: I didn't see it before `kwargs`, I was it was missing, then I realised there was another keyword argument before it :sweat_smile:

Reply:
I think we should add the dispatch for `FITS` as well, if we say that `FITS` is very broad and it's not good to narrow it down just for our case, we are already doing that when we are writing a `String` dispatch for `frame`. Thoughts?

Reply:
People don't really work with FITS objects directly, but with the HDUs instead. Plus this would just add another piece of complexity.